packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp

```diff
@@ -635,6 +635,13 @@ namespace MueLu {
 
           columns.resize(realnnz);
           numTotal = A->getNodeNumEntries();
+
+          // If the only element remaining after filtering is diagonal, mark node as boundary
+          for (LO row = 0; row < Teuchos::as<LO>(A->getRowMap()->getNodeNumElements()); ++row) {
+            if (rows[row+1]- rows[row] <= 1)
+              boundaryNodes[row] = true;
```

Comment:
Marking nodes as boundary when they are not boundary nodes seems wrong, I would rather have them tagged as `IGNORED` for aggregation, actually this is something that is achieved by running the `IsolatedNodeAggregationAlgorithm`.

Reply:
I agree that that would be cleaner. It looks like `IsolatedNodeAggregationAlgorithm` has been inactive since 2015 https://github.com/trilinos/Trilinos/commit/7e892ecb58513a6e6f9017f1e277915713a8f008#diff-4abf20d14b91dda5281d2427a0075a78ed80ca9a39342e1362fd3ccba6f741ef, and instead this functionality has been put into `CoalesceDrop`. Not sure why.